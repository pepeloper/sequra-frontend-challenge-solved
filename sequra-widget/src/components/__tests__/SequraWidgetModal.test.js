import { mount } from '@vue/test-utils'
import SequraWidgetModal from '../SequraWidgetModal.vue'


describe('SequraWidgetModal', () => {

  test('mount component', () => {
    expect(SequraWidgetModal).toBeTruthy()

    const wrapper = mount(SequraWidgetModal, {
      props: {
        creditAgreement: {
          "instalment_count": 3,
          "instalment_amount": { "string": "133,33 €" },
          "instalment_fee": { "string": "5 €" },
          "instalment_total": { "string": "138,33 €" },
        },
        isOpen: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})