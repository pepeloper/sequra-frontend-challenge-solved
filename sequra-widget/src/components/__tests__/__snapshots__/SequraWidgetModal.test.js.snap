// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SequraWidgetModal > mount component 1`] = `
"<div class=\\"sequra__widget__modal__overlay\\">
  <div class=\\"sequra__widget__modal\\" role=\\"dialog\\" aria-labelledby=\\"sequra-widget__label\\" aria-hidden=\\"false\\">
    <div class=\\"sequra__widget__modal__header\\">
      <p id=\\"sequra-widget__label\\">Fracciona tu pago</p>
    </div>
    <div class=\\"sequra__widget__modal__body\\">
      <ol>
        <li>Eliges \\"Fracciona tu pago\\" al realizar tu pedido y pagas solo la primera cuota</li>
        <li>Recibes tu pedido</li>
        <li>El resto de pagos se cargarán automáticamente a tu tarjeta</li>
      </ol>
      <p>¡Así de simple!</p>
      <p> Además, en el importe mostrado ya se incluye la cuota única mensual de 5 €/mes por lo que no tendrás ninguna sorpresa. </p>
    </div>
  </div>
</div>"
`;
